### PR TITLE
Applies assertion funcs to cache tests

### PR DESCRIPTION
**Pull request by rkapka**
**What type of PR is this?**

Other/Tests Refactoring

**What does this PR do? Why is it needed?**

applies `testutil/assert` and `testutil/require` to cache tests

**Which issues(s) does this PR fix?**

Part of #6590

**Other notes for review**

I don't know how to assert that a slice variable is `nil`. Trying `assert.Equal(t, ([]unit64)(nil), x)` results in
```
panic: runtime error: comparing uncomparable type []uint64 [recovered]
        panic: runtime error: comparing uncomparable type []uint64
```

### CODE REVIEW

File: beacon-chain/cache/skip_slot_cache_test.go [hunk 1]
```diff
@@ -1,53 +1,34 @@
-package cache_test
```
Comment (terencechain) [inline beacon-chain/cache/skip_slot_cache_test.go:1]: Why revert this?

Comment (rkapka) [reply beacon-chain/cache/skip_slot_cache_test.go:1]: Out of all test files in the cache folder some where in the `cache` package and some in the `cache_test` package so I decided to unify it. I can change it back if you'd like.

Comment (terencechain) [reply beacon-chain/cache/skip_slot_cache_test.go:1]: We should use `pkg_test` whenever we can to mimic block box testing. But sometimes we have to use `pkg` in tests to gain access to the non-exported identifiers. It's for unit tests that require access to non-exported variables, functions, and methods.

`pkg_test` is generally preferred over `pkg` for tests. A good follow up tasks from this is to look at other tests to see if we can convert them to `pkg_test`

Comment (farazdagi) [reply beacon-chain/cache/skip_slot_cache_test.go:1]: Furthermore, please, try to keep PRs constrained to the single purpose -- here, you are updating test to use assertions, no further refactoring should make it into this PR. From time to time, our PRs will contain more changes than is warrant by PR's goal, but we are trying to keep them focused, as a general rule.

**Pull request merged**